apps/web/src/pages/liquid-staking/[contract].tsx

```diff
@@ -0,0 +1,170 @@
+import { useTranslation } from '@pancakeswap/localization'
+import { CardBody, Text, RowBetween, Box, Flex, Image } from '@pancakeswap/uikit'
+import { AppBody, AppHeader } from 'components/App'
+import CurrencyInputPanel from 'components/CurrencyInputPanel'
+import Page from 'views/Page'
+import { useCurrency } from 'hooks/Tokens'
+import { LightGreyCard } from 'components/Card'
+import { useRouter } from 'next/router'
+import { useEffect, useMemo, useState } from 'react'
+import { getFullDisplayBalance } from '@pancakeswap/utils/formatBalance'
+import BigNumber from 'bignumber.js'
+import { ChainId } from '@pancakeswap/sdk'
+import { useCurrencyBalance } from 'state/wallet/hooks'
+import { BIG_ZERO } from '@pancakeswap/utils/bigNumber'
+import AddToWalletButton from 'components/AddToWallet/AddToWalletButton'
+import { maxAmountSpend } from 'utils/maxAmountSpend'
+import { LiquidStakingFAQs } from 'views/LiquidStaking/components/FAQs'
+import useAccountActiveChain from 'hooks/useAccountActiveChain'
+import { LiquidStakingList } from 'views/LiquidStaking/constants/types'
+import { fetchLiquidStaking } from 'views/LiquidStaking/hooks/useLiquidStakingList'
+import StakeInfo from 'views/LiquidStaking/components/StakeInfo'
+import { useExchangeRate } from 'views/LiquidStaking/hooks/useExchangeRate'
+import LiquidStakingButton from 'views/LiquidStaking/components/LiquidStakingButton'
+
+const LiquidStakingStakePage = () => {
+  const { t } = useTranslation()
+  const router = useRouter()
+  const { account, chainId } = useAccountActiveChain()
+  const [stakeAmount, setStakeAmount] = useState('')
+  const [showPage, setShowPage] = useState(false)
+  const [selectedList, setSelectedList] = useState<null | LiquidStakingList>(null)
+
+  useEffect(() => {
+    const contract: string = (router?.query?.contract as string) ?? ''
+    const fetch = async () => {
+      const list = await fetchLiquidStaking(chainId)
```

Comment:
seems like the fetch has nothing to do with the contract. Can we move it outside of the effect? Only refetch everytime chain id changes? Right now it will refetch every time the contract changes

Reply:
done